src/PrestaShopBundle/Translation/Provider/TranslationFinder.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation\Provider;
+
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use Symfony\Component\Translation\Loader\XliffFileLoader;
+use Symfony\Component\Translation\MessageCatalogue;
+use Symfony\Component\Translation\MessageCatalogueInterface;
+
+/**
+ * Helper used to build a MessageCataloguer from xliff files
+ */
+class TranslationFinder
+{
+    const ERR_NO_FILES_IN_DIRECTORY = 1;
+    const ERR_DIRECTORY_NOT_FOUND = 2;
+
+    /**
+     * @param string|array $paths a list of paths when we can look for translations
+     * @param string $locale the Symfony (not the PrestaShop one) locale
+     * @param string|null $pattern a regular expression
+     *
+     * @return MessageCatalogue
+     *
+     * @throws FileNotFoundException
+     */
+    public function getCatalogueFromPaths($paths, $locale, $pattern = null)
+    {
```

Comment:
Maybe some parts of this method could have been extracted in two different methods, that would be called in it ?

**getTranslationFilesFromPaths**() et **getFileCatalogueFromTranslationFile**();

It could make this method easier to read, but this is not mandatory, maybe my suggestion is overkill... your call.

Reply:
example:

```php

function getTranslationFilesFromPaths($paths)
    {
        try {
            $translationFiles = $finder->files()->notName('index.php')->in($paths);
        } catch (\InvalidArgumentException $e) {
            throw new FileNotFoundException(
                sprintf(
                    'Could not crawl for translation files: %s',
                    $e->getMessage()
                ),
                self::ERR_DIRECTORY_NOT_FOUND,
                $e
            );
        }
        if (count($translationFiles) === 0) {
            throw new FileNotFoundException('There are no translation file available.', self::ERR_NO_FILES_IN_DIRECTORY);
        }
        
        return $translationFiles;
    }

```